From 970f4ad767cdf8f4a12e4e02f66167ba249d955c Mon Sep 17 00:00:00 2001
From: Jiebing Li <jiebing.li@intel.com>
Date: Fri, 1 Aug 2014 15:21:14 +0800
Subject: [PATCH 3/4] usb/dwc3-device: fix kernel panic in
 dwc3_endpoint_transfer_complete()

There is a possibility that dep->endpoint.desc is modified to NULL
by another thread, after the spinlock in dwc3_endpoint_transfer_complete()
is released for a while.

So need to check this pointer before it's accessed.

Change-Id: I7c6a7381ccec8cf9c16a2190ae2eee8bf5301088
Signed-off-by: Jiebing Li <jiebing.li@intel.com>
---
 drivers/usb/dwc3/gadget.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index fd977e74407e..d9b288da3db6 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -1906,7 +1906,7 @@ static int dwc3_cleanup_done_reqs(struct dwc3 *dwc, struct dwc3_ep *dep,
 			break;
 	} while (1);
 
-	if (usb_endpoint_xfer_isoc(dep->endpoint.desc) &&
+	if (dep->endpoint.desc && usb_endpoint_xfer_isoc(dep->endpoint.desc) &&
 			list_empty(&dep->req_queued)) {
 		if (list_empty(&dep->request_list))
 			/*
-- 
1.7.9.5

